nilearn/signal.py

```diff
@@ -806,12 +806,30 @@ def clean(
         signals -= Q.dot(Q.T).dot(signals)
 
     # Standardize
-    if (detrend and standardize == "psc") or (filter_type == "butterworth"):
-        # If the signal is detrended or filtered,
-        # the mean signal will be zero or close to zero. In this case,
-        # we have to know the original mean signal to calculate the psc.
+    if not standardize:
+        return signals
+    filtered_mean_check = (
+        np.abs(signals.mean(0)).mean() / np.abs(original_mean_signals).mean()
+        < 1e-1
+    )
+    if standardize == "psc" and filtered_mean_check:
+        # If the signal is detrended, the mean signal will be zero or close to
+        # zero. If signal is high pass filtered with butterworth, the constant
+        # (mean) will be removed. This is detected through checking the scale
+        # difference of the original mean and filtered mean signal. When the
+        # mean is too small, we have to know the original mean signal to
+        # calculate the psc to avoid weird scaling.
+        warnings.warn(
+            "After cleaning, the new mean of the signal is 1 factor of 10 "
+            "smaller than the original signal. This can obscure the scale of "
+            "the signal with percent signal change standardization. To avoid "
+            "the scale of the signal being obscured drassically, the original "
+            "signal mean has been added back for percent signal change "
+            "standardization. Please note that this might not always solve "
+            "the scale issue if the original signal has a small mean."
+        )
```

Comment:
I'm a bit puzzled here. under which conditions ca such a shrinkage of the mean signal occur ?

Reply:
If there is no reason to keep it I'd rather remove it.